proto/frequenz/api/dispatch/v1/dispatch.proto

```diff
@@ -281,6 +282,13 @@ message RecurrenceRule {
   // - Every 2 hours:
   //   freq = FREQUENCY_HOURLY
   //   interval = 2
+  // Valid values depend on the value of the `freq` field:
+  // - FREQUENCY_MINUTELY: 1 to 10_000 (~7 days)
+  // - FREQUENCY_HOURLY: 1 to 10_000 (~416 days)
+  // - FREQUENCY_DAILY: 1 to 10_000 (~27 years)
+  // - FREQUENCY_WEEKLY: 1 to 1_000 (19 years)
+  // - FREQUENCY_MONTHLY: 1 to 1_000 (83 years)
```

Comment:
Why do we have these 2 special cases? Why not 10.000 always?

Reply:
As already described in the PR, these are directly taken from https://github.com/fmeringdal/rust-rrule?tab=readme-ov-file#limitation-and-limits and they are simply the limits that the rrule library imposes (when used in "arbitrary" mode as opposed to the unsafe "crate" mode)

Reply:
Woops, sorry I missed that. Can't we update the server instead to use only one limit? It looks like unnecessary complexity, specially considering that years can be 10.000, what's the point of limiting months and weeks to less than that?

Reply:
I suppose we can have 1_000 for all? 
But I don't really see the issue, like, what extra complexity? 
If you don't want it to depend on the frequency in your code can just limit everything to 1000, probably enough for every sane usecase

Reply:
>  what's the point of limiting months and weeks to less than that?

I assume that the calculation is way harder on the CPU than with the others? like, finding the exact date based on a week-offset compared to a year-offset?

Reply:
The complexity of having a table to know which limit applies to what. I agree 1.000 for everything should work too, I'm fine with that too.

Reply:
OK, the library says that the limitation is used for [security](https://github.com/fmeringdal/rust-rrule/blob/main/SECURITY.md) reasons. I think we should have some limits too.

Another alternative is not specifying a limit, and instead leave that to the server and client implementation to decide. The client can just be prepared to deal with an error if creating something out of bounds (we should probably send the maximum allowed value in the error response). Also the client could chose to truncate this value if it is too big for security reasons, so even if the dispatch had, say 1000, but the client allows only 500, use 500 instead and emit some warning/error.

It really feels like the limit could be different for different systems, like a powerful cloud server could accept 10.000 but a restricted client in the edge might support less than that.

Reply:
Adjusted the cmoment

Reply:
I'm not very familiar with how errors are reported in gRPC, but if there is a way or it makes sense to specify how errors will be returned here, I think it would be useful. Like: `Returns a "OutOfBounds" error if it surpasses the server limit`, and ideally we should be able to attach some key-value data to the error to have something like "limit=<value>", so the user can take corrective action, otherwise you'll need to start doing a "binary search" for the limit making several requests.

@tiyash-basu-frequenz do we have any standards on how to specify errors?

Reply:
I suppose the thing to consider here is if the errors are supposed to be human-readable or machine-readable.

If human-readability is enough, then a textual error message is enough, e.g. "Interval exceeded maximum allowed value 10000".

If we need really machine-readability, then we need to design and document a new standard (probably for formatting the error message). Such a standard could contain specs for key=value pairs. We do not have such a standard so far, because it was never needed.

Reply:
Please stay calm, but I have to inform you that an "Out of Bounds" error [already exists](https://grpc.io/docs/guides/status-codes/#full-list-of-status-codes)

Reply:
If we were to design such a system, it would probably be best to use the grpc metadata to attach values

Reply:
Yes, what I mean is something machine readable, so you can retry with a value that you know will succeed as a programmer. Another option is to have a roc call to retrieve limits from the server, but that's a much bigger change.

And yea, I know about the out of bounds error. I meant attaching the limit as part of the error.

Reply:
How abut visiting this topic after v1.0 release of the current phase?